### PR TITLE
Turn into readonly instead of terminate()

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Do not kill ourself when logical error happens in some replicated table. Turn into read only instead.

Detailed description / Documentation draft:

.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
But this is controversial...

**Comment by alexey-milovidov** [comment]
BTW, what exception do you see?

**Comment by tavplubix** [comment]
> BTW, what exception do you see?

Part ... intersects ...

It's not completely fixed, there are still some issues that may lead to this exception.

**Comment by alexey-milovidov** [comment]
Let's avoid merging this PR and instead fix all the remaining issues with intersecting parts.

File: src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp [hunk 1]
```diff
@@ -606,7 +606,8 @@ int32_t ReplicatedMergeTreeQueue::pullLogsToQueue(zkutil::ZooKeeperPtr zookeeper
                 tryLogCurrentException(log);
                 /// If it fails, the data in RAM is incorrect. In order to avoid possible further corruption of data in ZK, we will kill ourselves.
                 /// This is possible only if there is an unknown logical error.
-                std::terminate();
+                /// Turn into readonly mode instead of std::terminate();
+                storage.restarting_thread.setReadonly();
```
[the remark this answers is not in the document]
Comment (amosbird) [reply src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp:610]: Done.

**Pull request closed without merging**